### DISCUSSION TITLE
ar/integrations/jupyterlab/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#19158] [Docs]
# ar/integrations/jupyterlab/

استكشف دليل التكامل الخاص بنا الذي يشرح كيف يمكنك استخدام JupyterLab لتدريب نموذج YOLO11 . سنغطي أيضًا الميزات والنصائح الرئيسية للمشاكل الشائعة.

https://docs.ultralytics.com/ar/integrations/jupyterlab/

**Comment by new-php** [comment]
"This guide seems helpful, but there are a few things that could be improved. The steps for training YOLO11 models in JupyterLab are unclear in some parts, and it would be great to see more detailed explanations, especially for beginners. I also noticed that some important configurations for training weren’t mentioned, which could cause confusion for users trying to set up the environment correctly.  

It would be really helpful to include sample images or screenshots to clarify key steps. Otherwise, new users might struggle with understanding the exact flow.  

For more details, check out:  
 [JupyterLab Installation Guide](https://docs.vultr.com/how-to-set-up-a-jupyterlab-environment-on-ubuntu-22-04)  

This version should come across as more critical and focused on the potential issues. Feel free to adjust further!

**Reply by glenn-jocher** [reply]
@new-php thank you for your feedback! We'll review the guide to improve clarity in training steps and configurations. For detailed YOLO11 training examples in JupyterLab, see our [FAQ section](https://docs.ultralytics.com/integrations/jupyterlab/#faq) with code snippets and best practices. Visual enhancements are planned for future updates.

**Comment by UltralyticsAssistant** [comment]
👋 Hello, thank you for sharing your interest in integrating YOLO11 with JupyterLab 🚀! It's great to see you exploring our tools. For new users, we recommend checking out our [Docs](https://docs.ultralytics.com/) that cover a variety of usage examples, including [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) instructions.

If you’re encountering any issues or have questions about this integration, please provide more details here. For 🐛 Bug Reports, ensure you include a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/). For ❓ Questions about custom training or integrations, please share relevant context such as error logs, configuration files, or specific challenges you're facing.

Join the Ultralytics community to connect further:
- Chat with us in real-time on [Discord](https://discord.com/invite/ultralytics) 🎧
- Dive into detailed discussions on [Discourse](https://community.ultralytics.com/)
- Share insights on our [Subreddit](https://reddit.com/r/Ultralytics)

## Upgrade

For the best experience, ensure you are using the latest version of the `ultralytics` package. Upgrade it alongside the [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

Explore several verified environments to run YOLO11 efficiently, including:
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See our [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. Refer to our [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. Follow the [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

Check out our continuous integration (CI) testing badge to verify the latest CI status:
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>  
If this badge is green, our CI tests are successfully verifying YOLO's compatibility with macOS, Windows, and Ubuntu environments for all supported [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/).

Lastly, don’t forget to reference our dedicated guide for working with JupyterLab here 📚: [JupyterLab Integration Guide](https://docs.ultralytics.com/ar/integrations/jupyterlab/). An Ultralytics engineer will follow up soon to assist further!
